jsonschema_gentypes/validate.py

```diff
@@ -0,0 +1,155 @@
+) -> Tuple[List[str], Dict[str, Any]]:
+    """
+    Vaildate the YAML, with him JSON schema.
+
+    Arguments:
+        filename: Name used to generate the error message
+        data: The data structure to be validated (should be loaded with ruamel.yaml to have the lines number)
+        schema: The loaded JSON schema
+        default: Should fill the data with the default provided in the JSON schema
+    """
+    schema_ref = schema.get("$schema", "default")
+    schema_match = re.match(r"https?\:\/\/json\-schema\.org\/(.*)\/schema", schema_ref)
+    Validator = {  # noqa: N806 # variable 'Validator' in function should be lowercase
+        "draft-04": jsonschema.Draft4Validator,
+        "draft-06": jsonschema.Draft6Validator,
+        "draft-07": jsonschema.Draft7Validator,
+    }.get(
+        schema_match.group(1) if schema_match else "default",
+        jsonschema.Draft7Validator,
+    )
+    if default:
+        Validator = _extend_with_default(  # noqa: N806 # variable 'Validator' in function should be lowercase
+            Validator
+        )
+
+    validator = Validator(schema)
+
+    def format_error(error: jsonschema.exceptions.ValidationError) -> str:
+        position = filename
+
+        if hasattr(error.instance, "lc"):
+            position = f"{filename}:{error.instance.lc.line + 1}:{error.instance.lc.col + 1}"
+        else:
+            curent_data = data
+            parent = None
+            if hasattr(curent_data, "lc"):
+                parent = curent_data
+            for path in error.path:
+                curent_data = curent_data[path]
+                if hasattr(curent_data, "lc"):
+                    parent = curent_data
+            if parent is not None:
+                position = f"{filename}:{parent.lc.line + 1}:{parent.lc.col + 1}"  # type: ignore
+
+        return (
+            f" - {position} "
+            f'({".".join([str(i) for i in error.path] if error.path else "/")}): '
+            f"{error.message} (rule: "
+            f'{".".join([str(i) for i in error.schema_path] if error.schema_path else "/")})'
+        )
+
+    return sorted([format_error(e) for e in validator.iter_errors(data)]), data
+
+
+def load(
+    filename: str,
+    schema: Dict[str, Any],
+    default: bool = True,
+    exit_on_error: bool = True,
+) -> Any:
+    """
+    Load and vaildate the YAML, with him JSON schema.
+
+    Arguments:
+        filename: Name of the file that should be loaded
+        schema: The JSON schema
+        default: Should fill the data with the default provided in the JSON schema
+        exit_on_error: Should exit on error
+    """
+    with open(filename) as data_file:
+        errors, data = validate(filename, ruamel.yaml.round_trip_load(data_file), schema, default)  # type: ignore
+
+    if errors:
+        LOG.error("The config file is invalid:\n%s", "\n".join(errors))
+        if exit_on_error:
+            sys.exit(1)
```

Comment:
I would prefer a raise here (API not CLI). Caller could handle it as he wants.

Reply:
Then I also remove the exit_on_error...